src/dune_pkg/opam_repo.mli

```diff
@@ -25,13 +25,12 @@ module Source : sig
 
   val commit : t -> Commitish.t option
   val url : t -> string
-  val of_opam_url : OpamUrl.t -> t Fiber.t
-  val to_string : t -> string
```

Comment:
I removed `to_string` from the public API. It's not clear what this string representation is good for (serialization? pretty printing?. We have better types for things like that `Pp.t`, `Dune_sexp.Encoder.t`, etc.